clang/test/CodeGenSYCL/sycl-cuda-host-device-functions.cu

```diff
@@ -0,0 +1,89 @@
+// RUN: %clang_cc1 -fsycl-is-host -internal-isystem %S/Inputs -sycl-std=2020 -emit-llvm %s -o - | FileCheck %s -check-prefix CHECK-HOST
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -emit-llvm %s -o - | FileCheck %s -check-prefix CHECK-DEV
+
+// This tests
+// - if a dummy __host__ function (returning undef) is generated for every
+//   __device__ function without a host counterpart in sycl-host compilation;
+// - if __host__ or __device__  functions are generated correctly depending
+//   on the compilation;
+// - if __host__ and __device__ functions in sycl-device compilation are
+//   generated with weak_odr linkage for avoiding multiple declaration of
+//   the same function due to the cuda-device compilation.
+
+#include "../CodeGenCUDA/Inputs/cuda.h"
+#include "sycl.hpp"
+
+__host__ int fun0() { return 7; }
+__device__ int fun0();
+
+// CHECK-HOST: define dso_local noundef i32 @_Z4fun0v()
+// CHECK-HOST: ret i32 7
```

Comment:
I think we must check that `ret i32 7` belongs to `fun0` definition and not `fun6` definition.
There are a few ways to achieve this, but probably the easiest one is to return a unique constant value.

Reply:
Done.